### PR TITLE
Python API for `pairwise_polygon_distance`

### LINKED ISSUE #1054: Python API Polygon-Polygon Distance

### PR DESCRIPTION
**Pull request by isVoid**
## Description

This PR adds python API for `pairwise_polygon_distance`

depends on #1073 

closes #1054 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cuspatial/cuspatial/core/spatial/distance.py [hunk 1]
```diff
@@ -555,6 +556,59 @@ def pairwise_linestring_polygon_distance(
...
+        The (multi)polygons to compute the distance from.
+    Returns
+    -------
+    distance : cudf.Series
+
+    Notes
+    -----
+    The input `GeoSeries` must contain a single type geometry.
+    For example, `polygons1` series cannot contain both points and
+    polygons.
```
Comment (thomcom) [inline python/cuspatial/cuspatial/core/spatial/distance.py:584]: Seems redundant - polygons1 and polygons2 must contain only polygons.

File: python/cuspatial/cuspatial/core/spatial/distance.py [hunk 2]
```diff
@@ -555,6 +556,59 @@ def pairwise_linestring_polygon_distance(
...
+    if len(polygons1) == 0:
+        return cudf.Series(dtype=polygons1.lines.xy.dtype)
+
+    if not contains_only_polygons(polygons1):
+        raise ValueError("`polygons1` array must contain only polygons")
+
+    if not contains_only_polygons(polygons2):
+        raise ValueError("`polygons2` array must contain only polygons")
+
+    # Handle slicing and aligns in geoseries
```
Comment (thomcom) [inline python/cuspatial/cuspatial/core/spatial/distance.py:603]: This line is now deprecated I think?

File: python/cuspatial/cuspatial/tests/spatial/distance/test_pairwise_polygon_distance.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+def test_point_polygon_geoboundaries(naturalearth_lowres):
+    N = 50
+
+    lhs = naturalearth_lowres.geometry[:N].reset_index(drop=True)
+    rhs = naturalearth_lowres.geometry[N : 2 * N].reset_index(drop=True)
+    expect = lhs.distance(rhs)
+    got = cuspatial.pairwise_polygon_distance(
+        cuspatial.GeoSeries(lhs), cuspatial.GeoSeries(rhs)
+    )
+    assert_series_equal(cudf.Series(expect), got)
```
Comment (thomcom) [inline python/cuspatial/cuspatial/tests/spatial/distance/test_pairwise_polygon_distance.py:149]: Seems like you might have a couple of simple tests here, like `self.distance(self)` and `polygon.distance(touching)` and a pair where the distance `== 1`.

**Review by thomcom — CHANGES_REQUESTED** [review]

File: python/cuspatial/cuspatial/core/spatial/distance.py [hunk 4]
```diff
@@ -555,6 +556,59 @@ def pairwise_linestring_polygon_distance(
...
+    Notes
+    -----
+    The input `GeoSeries` must contain a single type geometry.
+    For example, `polygons1` series cannot contain both points and
+    polygons.
+
+    Examples
+    --------
+    Compute distance between a point and a polygon:
+    """
```
[the remark this answers is not in the document]
Comment (isVoid) [reply python/cuspatial/cuspatial/core/spatial/distance.py:609]: added

File: python/cuspatial/cuspatial/tests/spatial/distance/test_pairwise_polygon_distance.py (hunk 3 above)
Comment (isVoid) [reply python/cuspatial/cuspatial/tests/spatial/distance/test_pairwise_polygon_distance.py:152]: added

**Review by thomcom — APPROVED** [review]

File: python/cuspatial/cuspatial/core/spatial/distance.py [hunk 5]
```diff
@@ -555,6 +556,78 @@ def pairwise_linestring_polygon_distance(
     )
 
 
+def pairwise_polygon_distance(polygons1: GeoSeries, polygons2: GeoSeries):
+    """Compute distance between pairs of (multi)polygons and (multi)polygons
```
Comment (harrism) [inline python/cuspatial/cuspatial/core/spatial/distance.py:560]: I would use an explicit pair notation instead for clarity and brevity:

```suggestion
    """Compute distance between (multi)polygon-(multi)polygon pairs."
```

hyphenate the pair relationship, and be consistent across APIs. So for `pairwise_linestring_polygon_distance` you would use "distance between linestring-polygon pairs".

File: python/cuspatial/cuspatial/core/spatial/distance.py [hunk 6]
```diff
@@ -555,6 +556,78 @@ def pairwise_linestring_polygon_distance(
...
+    The distance between a (multi)polygon and a (multi)polygon
+    is defined as the shortest distance between every edge of the
+    (multi)polygon pair. If the multipolygon and multipolygon intersects,
+    the distance is 0.
```
Comment (harrism) [inline python/cuspatial/cuspatial/core/spatial/distance.py:564]:
```suggestion
    The distance between two (multi)polygons is defined as the shortest
    distance between any edge of the first (multi)polygon and any edge
    of the second (multi)polygon. If two (multi)polygons intersect, the 
    distance is 0.
```

**Review by harrism — APPROVED** [review]
Approve, just some doc improvements.

**Pull request merged**